### PR TITLE
Move BaseDialog, subclasses and their controllers to dialog subfolder

### PR DESCRIPTION
**Pull request by Abyss777**
Moved Login, Register, Command, DeviceDistance and all object edit dialog. They all children or grandchildren BaseDialog.

I'm just thinking, may be we should remove `Dialog` from names because it is in path already, what do you think?

### CODE REVIEW

File: web/app/view/dialog/ReportConfigController.js [hunk 1]
```diff
@@ -16,7 +16,7 @@
  * along with this program. If not, see <http://www.gnu.org/licenses/>.
  */
 
-Ext.define('Traccar.view.ReportConfigController', {
+Ext.define('Traccar.view.dialog.ReportConfigController', {
```
Comment (tananaev) [inline web/app/view/dialog/ReportConfigController.js]: I think this class should be renamed to `ReportConfigDialogController`. Also please check that all others match naming convention. Controller should have the same name as the view plus the "Controller" suffix.

**Comment by tananaev** [comment]
Agree about the removal of "Dialog" from the names, if it's already in the path.

**Comment by Abyss777** [comment]
Removed `Dialog` from names and aliases.
Two controller names became consistent to their dialog names: `DeviceDistanceController` and `ReportConfigController`

**Comment by tananaev** [comment]
Merged, thanks. Everything looks much cleaner now.

**Comment by Abyss777** [comment]
I have two more things to group and move in mind:
- `grid.Panel`s with `BasePermissionsController` used for permissions `User*` and `*Geofences`, move to subfolder "permissions" (or something shorter?)
- `grid.Panel`s with `EditTollbar` used for editing objects except `Devices`, move to some folder, but do not know how to call it.

What do you think?

**Comment by tananaev** [comment]
- Agree with `permissions` subfolder. It sounds good to me.
- Maybe call it just `edit` or something like this.

**Pull request merged**